app/services/sso_service.rb

```diff
@@ -6,6 +6,10 @@
 class SSOService
   include SentryLogging
   include ActiveModel::Validations
+  attr_reader :auth_error_code
+  AUTH_ERRORS = { 'Subject did not consent to attribute release' => '001',
+                  'Current time is on or after NotOnOrAfter condition' => '002',
+                  'Current time is earlier than NotBefore condition' => '003' }.freeze
```

Comment:
two things: 

1. There could be more than one error, see in AuthFailHandler how multi_errors gets returned. How will we handle that? Possibly consider a generic error around SAML, distinct from generic error related to user object persistence.

2. We should at least include a generic error message. The FE will not be able to distinguish a generic SAML error from a generic User persistence error otherwise.

Reply:
Maybe instead of the codes you are using, we should calculate a Checksum based on the message and use that as the code.

Reply:
The idea around using a checksum would be that the FE (tagging @U-DON) could set a warning or info for unidentified codes that get returned (add extra_context with the checksum that is unmapped), that way we can come up with unique copy for those in the future.

Reply:
actually, come to think of it, logging those on the FE won't be much good. 😂 we will just have a checksum, and won't know what caused it. @dhall can you add additional logging so we can surface errors returned that the FE would not have mappings for?

Reply:
I wonder if maybe we don't want to use code, and just provide the message exactly as is? 😬 that seems risky unless we can html escape it somehow to prevent XSS though.

Reply:
I think I'd rather avoid the tight coupling that would come with passing the message text. If something changes with the wording, we are updating two code bases instead of handling it as an error scenario that the FE doesn't need to know what the backend error message is, since that's not the message that is displayed to a user.

Reply:
We're going to have that tight coupling one way or the other. If the message changes it will not match your codes, and hence it will end up being the generic error message. ;)